app/controllers/api/v1/documents.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'doorkeeper/grape/helpers'
+
+module API
+  module V1
+    class Documents < Grape::API
+      helpers Doorkeeper::Grape::Helpers
+
+      before do
+        doorkeeper_authorize!
+
+        def current_account
+          Account.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+        end
+      end
+
+      desc 'Account related routes'
+      resource :documents do
+        desc 'Return current user documents list'
+        get '/' do
+          current_account.documents.as_json(only: %i[upload doc_type doc_number doc_expire])
+        end
+
+        desc 'Upload a new document for current user'
+        params do
+          requires :doc_expire, :doc_type, :doc_number, :upload
+        end
+
+        post '/' do
+          doc = Document.new(declared(params).merge(account: current_account))
+          return error!(doc.errors.full_messages.to_sentence, 400) unless doc.save
```

Comment:
Please write this in 5 lignes of code,
it's not about the least amount of lines
you should use a begin rescue block

is declared(params).merge(account: current_account)
what happens if the account exists
what happens if params are nil or empty?

what do you return in case of success?

Do you even know?

write me a clear structure because I have no idea

```ruby
post '/' do
  begin
    parse params
    validate params
    save params
    return success
  rescue error
     log_error
     return formatted error from helper
  end
end
```